package.json

```diff
@@ -12,6 +12,10 @@
     ".": {
       "import": "./dist/esm/index.js",
       "require": "./dist/cjs/index.js"
+    },
+    "./migrations": {
+      "import": "./dist/esm/migrations.js",
+      "require": "./dist/cjs/migrations.js"
```

Comment:
alternative suggestion:
other than file paths, `"import"` and `"require"` can also be objects, containing `"types"`  and `"default"`.
could leverage those declarations here, which might make `dist-fix.js` obsolete?

Reply:
Interesting! 🤔 

I think we'd still want to alter how `tsup` outputs and have a folder for each still. What are the downsides?

Reply:
> have a folder for each still

_intuitively_ I'd agree. (read: me wanting to organize and put different things into different boxes)

at the same time, that requirement may be obsolete?
- these days `.mjs` file extensions seem widely supported.
- the specs allow for arbitrary paths via `"exports"` & co-located type def files with matching extensions. wanting an "esm"/"cjs" folder seems like custom build decision that doesn't really impact lib users(?)
- I'd trust tools like `tsup` to know better than my own intuition; there's likely a reason outputting separate folders is non-default and moved behind the aptly named `--legacy-output` flag. ¯\\\_(ツ)\_/¯

> What are the downsides?

omitting `--legacy-output` would generate ESM files using `.mjs` file extensions. I haven't worked with any bundler/environment in some time that does _not_ understand that extension. theoretically some (archaic?) environment might not understand `.mjs`, but for that to be relevant they'd also need to somehow explicitely want to load the `.mjs` file over the co-located CJS `.js` file.

(Hypothesis: Could be that this was more relevant at a time when ESM was still competing with CJS _and_ IIFE. Back then, I don't think we had extensions to differentiate between those, or they weren't as well established yet; and we certainly did not have `"exports"` in `package.json` back then.)

Reply:
_(Personally I'd even be in favor of leaving CJS in the dust. But I understand that, for now, it may still be controversial forcing that decision as lib author.)_

Reply:
> See if anyone complains.

nice - Test-In-Prod is my middle name 👋 changes pushed!

I've made sure to keep the `/// <reference .../>` injection (now also widened to the CSJ files).
not fully sure how to handle the chunk files (or they need any handling) for deno, so I've omitted them for now. if problematic, can also config tsup to not output chunks I believe.

while I was at it, I've also moved the dist-fix script to ESM in a separate comit - hope that's cool? no real reason other than cLeAnUp. happy to revert if CJS is preferred.